### PR TITLE
upgrade `validate_inputs` function

### PR DESCRIPTION
**Pull request by chlebowa**
Rewrites `validate_inputs*` functions into a single function.
`validate_inputs` will accept an arbitrary number of validators passed directly or as a list, with different results.

### CODE REVIEW

File: tests/testthat/test-get_rcode.R [hunk 1]
```diff
@@ -18,7 +18,7 @@ testthat::test_that("get_rcode returns code from chunks at the end", {
   teal.code::chunks_push(id = "test", chunks = ch, quote(a <- 1))
 
   r_code_from_chunks <- strsplit(get_rcode(chunks = ch), "\n")[[1]]
-  r_code_from_header <- strsplit(sprintf("\n\n%s\n\na <- 1", paste(get_rcode_header(), collapse = "\n")), "\n")[[1]]
+  r_code_from_header <- strsplit(sprintf("\n\n%s\na <- 1", paste(get_rcode_header(), collapse = "\n")), "\n")[[1]]
```
Comment (nikolas-burkoff) [inline tests/testthat/test-get_rcode.R:21]: Lets not change this as it's being done here https://github.com/insightsengineering/teal/pull/800 ?

**Review by nikolas-burkoff — COMMENTED** [review]
Is it possible to allow the case:

`validate_inputs( list(head1 = iv1, head2 = list(iv2, iv3))` because then this case:

`validate_inputs(iv1, iv2, iv3)` can initially create  `x <- list(list(iv1, iv2, iv3)); names(x) <- header` and then both that case and `validate_inputs(list(h1 = iv1, h2 = iv2))` can follow exactly the same logic for the rest  of the function.

If that's not possible then nevermind - looks good

**Comment by chlebowa** [comment]
> validate_inputs(iv1, iv2, iv3) can initially create x <- list(list(iv1, iv2, iv3)); names(x) <- header and then both that case and validate_inputs(list(h1 = iv1, h2 = iv2)) can follow exactly the same logic for the rest of the function.

It's not entirely the same because the "direct method" groups all validator messages under one header and the "list method" gives them separate headers. `structure(list(list(...)), names = header)` followed by `unlist(vals)` will give the same header twice:
image

Or did I misunderstand you suggestion?

**Review by nikolas-burkoff — APPROVED** [review]

File: R/validate_inputs.R [hunk 2]
```diff
@@ -9,30 +9,28 @@
...
 #'
-#' `validate_inputs` accepts an arbitrary number of `InputValidator`s
-#' and prints all messages together, adding one (optional) header.
-#' `validate_inputs_segregated` accepts a list of `InputValidator`s
-#' and prints messages grouped by validator. If elements of `validators` are named,
-
-#' the names are used as headers for their respective message groups.
-#'
+#' The `...` argument accepts any number of `InputValidator` objects
+#' or an (possibly infinitely) nested list of such objects.
```
Comment (nikolas-burkoff) [inline R/validate_inputs.R:18]: I would remove (possibly infinitely) here as it's a bit much...

File: R/validate_inputs.R [hunk 3]
```diff
@@ -101,79 +99,81 @@
...
+}
 
-  if (!all(vapply(validators, validator_enabled, logical(1L)))) {
-    logger::log_warn("Some validators are disabled and will be omitted.")
-    validators <- Filter(validator_enabled, validators)
-  }
+#' @keywords internal
+# test if an InputValidator object is enabled
+# returns logical of length 1
+validator_enabled <- function(x) {
```
Comment (nikolas-burkoff) [inline R/validate_inputs.R:129]: I would put the github issue from shinyvalidate as a code comment here

**Review by nikolas-burkoff — APPROVED** [review]

**Pull request merged**